### DISCUSSION TITLE
Get test results of a run

### DISCUSSION THREAD

**Discussion by blazmrak** [spockframework/spock#1487] [Ideas]
It might be a noob question/suggestion, but from what I understand, the only way to get test results now is to implement them yourself - implement the IGlobalExtension and add spec listeners, etc. Does any extension exist already that returns structured test data? Something like `{specs: [{...specStuff, features: [{...featureStuff, iterations: [{iterationStuff...,}]}]}]}`, which can be easily used to make reports.

**Comment by alopukhov** [comment]
Spock 2 runs on JUnit 5 Platform. Junit platform can generate junit4 format xml reports https://junit.org/junit5/docs/current/user-guide/#junit-platform-reporting
Enabling it depends on the way you are starting your tests.
For junit console launcher it may be enabled with parameter  `--reports-dir` (https://junit.org/junit5/docs/current/user-guide/#running-tests-console-launcher-options)
For gradle build tool it's enabled by default (xml report will be written in each modules' <buildDir>/test-results/....)
Junit4xml format is widely used and compatible with many tools.
